prov/tcp/src/tcpx_progress.c

```diff
@@ -434,8 +434,16 @@ void tcpx_progress(struct util_ep *util_ep)
 
 static int tcpx_try_func(void *util_ep)
 {
-	/* nothing to do here. When endpoints
-	   have incoming data, cq drives progress*/
+
+	struct tcpx_ep *ep;
+	ep = container_of(util_ep, struct tcpx_ep, util_ep);
+
+	if (!slist_empty(&ep->tx_queue))
+		return -FI_EAGAIN;
+
+	if (!ep->cur_rx_entry)
+		return -FI_EAGAIN;
+
```

Comment:
I don't think this is the correct fix.  It will end up spinning the thread if there's any queued work to do, even if the underlying socket is blocked.  And the underlying socket can be blocked because the remote side isn't reading data.

Reply:
Thanks for the review, @shefty.
Sorry but I haven't had a chance to work on the patch so far.